infer/src/pulse/PulseExecutionDomain.ml

```diff
@@ -102,6 +108,170 @@ let pp fmt exec_state =
+      | LatentAbortProgram a ->
+          AbductiveDomain.Summary.get_path_condition a.astate
+      | LatentInvalidAccess a ->
+          AbductiveDomain.Summary.get_path_condition a.astate
+      | InfiniteProgram astate ->
+          AbductiveDomain.get_path_condition astate
+      | ExceptionRaised astate ->
+          AbductiveDomain.get_path_condition astate
+      | ContinueProgram astate ->
+          AbductiveDomain.get_path_condition astate
+      | LatentSpecializedTypeIssue {astate; _} ->
+          AbductiveDomain.Summary.get_path_condition astate
+    in
+    (* Used when the workset is not empty *)
+    let rec record_pathcond ws kl : int =
+      let cmp_four (a, b, c, d) (e, f, g, h) =
+        if phys_equal a e && phys_equal b f && phys_equal c g && phys_equal d h then true else false
+      in
+      match ws with
+      | [] ->
+          -1
+      | (hd, idx) :: tl -> (
+          let cond = extract_pathcond hd in
+          let pathcond = Formula.extract_path_cond cond in
+          let termcond = Formula.extract_terminal_atoms cond in
+          let termcond2 = Formula.extract_terminal_terms cond in
+          let key = (cfgnode, termcond, pathcond, termcond2) in
+          let dl_ws = get_widenstate () in
+          match dl_ws with
+          | None ->
+              -1
+          | Some ws -> (
+              let prevstate = Stdlib.Hashtbl.find_opt ws key in
+              match prevstate with
+              | None ->
+                  let _ = add_widenstate key in
+                  record_pathcond tl (kl @ [key])
+              | Some _ -> (
+                match
+                  ( Formula.set_is_empty termcond
+                  , Formula.map_is_empty pathcond
+                  , Formula.termset_is_empty termcond2 )
+                with
+                | true, true, true ->
+                    idx
+                | _ ->
+                    let test = List.mem ~equal:cmp_four kl key in
+                    if test then -2 else idx ) ) )
+    in
+    let (repeated_wsidx : int) = if phys_equal same true then -1 else record_pathcond workset [] in
+    let create_infinite_state (hd : t) : t =
+      (* Only create infinite state from a non-error state that is not already an infinite state *)
+      match hd with
+      | ExceptionRaised astate ->
+          InfiniteProgram astate
+      | ContinueProgram astate ->
+          InfiniteProgram astate
+      | AbortProgram astate ->
+          AbortProgram astate
+      | ExitProgram astate ->
+          ExitProgram astate
+      | LatentAbortProgram a ->
+          LatentAbortProgram a
+      | LatentInvalidAccess a ->
+          LatentInvalidAccess a
+      | InfiniteProgram astate ->
+          InfiniteProgram astate
+      | LatentSpecializedTypeIssue astate ->
+          LatentSpecializedTypeIssue astate
+    in
+    let create_infinite_state_and_print (state_set : t list) (idx : int) (case : int) =
+      let _ = case in
```

Comment:
you can use `_case` to avoid warnings about unused values
```suggestion
    let create_infinite_state_and_print (state_set : t list) (idx : int) (_case : int) =
```